### DISCUSSION TITLE
I've made an easy peasy script for llama.cpp

### DISCUSSION THREAD

**Discussion by mounta11n** [ggml-org/llama.cpp#1544] [Show and tell]
Hey guys I want to share with you a script that I wrote and made a lot of work easier for me. It automatically finds your models, simplifies the option selection for you, generates the necessary "./main" command for llama.cpp, automatically generates and shows you the help file and in the worst case you can even interact with ChatGPT.. yay..

[![asciicast](https://asciinema.org/a/586215.svg)](https://asciinema.org/a/586215)
https://gitlab.com/mountaiin/dialema
https://codeberg.org/mountain/dialema

This might be interesting for those who have like me a million model files and who are tired of searching, symlinking or copying and pasting directory paths after typing "./main -m".
(in my case for example, i have model files on one internal and two external drives, so they are not centralized at one place)

The script opens a "dialog" text user interface from where you can comfortably set options and select a model from the models menu which tries to automatically find all model files on your computer or on external drives. Adapt this to your needs. In my case I am looking for .bin files that are at minimum 1000 MB.

This work is still in progress (and not fast progress, since this is kind of edcucational and learning by doing stuff), but I hope to finish implementing features like saving and loading config files or monitoring, saving and resuming chat sessions in the next days (since this is almost finished and implemented).

I'm not a professional programmer, but have only a few months ago started to deal with a few languages and with Bash scripting. I have become a big bash scripting fan. This script was originally planned for my personal use and as an education, now I thought that this would be a way to give something back to this cool community.

I don't know the git procedures very well yet, but I offer it on codeberg and here and I would be very happy if someone who is used to git and/or bash can tell me where to change or add to the code or even modify the script by him-/herself.

In some parts of the code or in the comments there are still some things in German and I haven't translated them yet, but I made sure that I translated the basic and important things in English for now. The rest comes later.

**Comment by thesimpleone** [comment]
Thats awesome.  Going to check it out
